tests/admin_filters/tests.py

```diff
@@ -1038,6 +1039,23 @@ def test_filter_with_failing_queryset(self):
         with self.assertRaises(ZeroDivisionError):
             modeladmin.get_changelist_instance(request)
 
+    def test_filter_for_change_list_form(self):
+        modeladmin = BookAdmin(Book, site)
+        tests = (
+            ({'q': '\x00'}, 'Null characters are not allowed'),
+        )
+
+        def get_request(kwargs):
+            request = self.request_factory.get('/', kwargs)
+            request.user = self.alfred
+            request._messages = CookieStorage(request)
+            modeladmin.get_changelist_instance(request)
+            return request
+
+        for case, error in tests:
+            with self.subTest(case=case):
+                self.assertIn(error, str(next(iter(get_request(case)._messages))))
```

Comment:
Rather than `str(next(iter(...` maybe: 

```
                messages = [m.message for m in request._messages]
                self.assertEqual(1, len(messages))
                self.assertIn(error, messages[0])
```